### DISCUSSION TITLE
The recv packet count is related to the CPU perfomance?

### DISCUSSION THREAD

**Discussion by Ncerzzk** [svpcom/wfb-ng#331] [General]
I'm doing some experiment by libpcap, and use a esp32 to inject wfb packets.
Then I use wireless card AR9271 on my PC and a SBC(orangepi3 ). The distance between PC/SBC with esp32 is the same.
It seems that the PC could recv about 4000 - 5000 packets in 10s ,
while SBC could only recv about 3000 - 4000 packets in 10s.

PS:
(I have tried to on and off the immediate_mode, the result changes little)
(Ar9271 is connect with PC/SBC on USB3.0 interface) 

So It seems that it may lost some packets if the cpu is not enough fast to handle?

**Comment by svpcom** [comment]
What is cpu load during receiving? Also SBC can generate noise in RF for example.

**Comment by Ncerzzk** [comment]
> What is cpu load during receiving? Also SBC can generate noise in RF for example.

Thank you, very light load(no other application running), what's the "noise" generated by SBC?

My test is:
libpcap running at block mode:

```
loop 10s{
      pcap_get_next_packet();
      cnt +=1;
}
```

I'm worried this loss happened in the Kernel network stack, in which we have few ways to optimize it.

**Reply by svpcom** [reply]
For example USB2 frequency is ~ 480mhz, USB3: ~5GHz

**Reply by Ncerzzk** [reply]
My test work on 2.4Ghz channel, so it seems no matters?
Have you met similar issue or compared the packets between RaspberryPi or PC?

**Reply by svpcom** [reply]
No, in my case both RPI3 and PC get 99% of packets  (packet loss is less that 1% and compensated via FEC)

**Reply by Ncerzzk** [reply]
Thank you. 
I found the cause, the onboard wifi module in SBC would affect the  wifibroadcast, though they are running at different channel(4 and 11). When I set the onboard wifi interface down, it behavior normally.
